automatic/git.install/tools/helpers.ps1

```diff
@@ -0,0 +1,48 @@
+function Get-InstallKey()
+{
+    $keyName = 'Git_is1'
+    $installKey = if ((Get-ProcessorBits 64) -and $env:chocolateyForceX86 -ne 'true') {
+             "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$keyName"
+    } else { "HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall\$keyName" }
+
+    $userInstallKey = "HKCU:\Software\Microsoft\Windows\CurrentVersion\Uninstall\$keyName"
+    if (Test-Path $userInstallKey) { $installKey = $userInstallKey }
+
+    mkdir $installKey -ea 0 | Out-Null
+    $installKey
+}
+
+function Set-InstallerRegistrySettings( [HashTable]$pp )
+{
+    $installkey = Get-InstallKey
+    $ino = "Inno Setup CodeFile:"
+
+    if ($pp.GitOnlyOnPath)         { New-ItemProperty $InstallKey -Name "$ino Path Option"          -Value "Cmd"            -Force }
+    if ($pp.GitAndUnixToolsOnPath) { New-ItemProperty $InstallKey -Name "$ino Path Option"          -Value "CmdTools"       -Force }
+    if ($pp.WindowsTerminal)       { New-ItemProperty $InstallKey -Name "$ino Bash Terminal Option" -Value "ConHost"        -Force }
+    if ($pp.NoAutoCrlf)            { New-ItemProperty $InstallKey -Name "$ino CRLF Option"          -Value "CRLFCommitAsIs" -Force }
+}
+
+function Get-InstallComponents( [HashTable]$pp )
+{
+    $res = "icons", "assoc", "assoc_sh"
+
+    if ($pp.NoShellIntegration) {
+        Write-Host "Parameter: no git shell integration"
+    } else { $res += "ext", "ext\shellhere", "ext\guihere" }
+
+    # Make our install work properly when running under SYSTEM account (Chef Cliet Service, Puppet Service, etc)
+    $isSystem = ([System.Security.Principal.WindowsIdentity]::GetCurrent()).IsSystem
+    if ( !$isSystem ) { $res += "icons\quicklaunch" }
+
+    if ($res.Length -eq 0) { return }
+    '/COMPONENTS="{0}"' -f ($res -join ",")
```

Comment:
@ferventcoder 

I think its 100+ lines of code more for the same thing. While LOC is not particularly good measure for code quality that code is basically unreadable and breaks number of paradigms of good programing. This is like PHP stuff - lets put everything in a single file and hope that nobody will ever change it. 

Contrary to that, refactored code has a benefit of
- SRP (single responsibility principle)
- DRY (do not repeat)
- Progressive disclosure - anybody looking the code will see critical phases - set parameters, stop agent , run installer, set path - on a single screen. If you want to know particular details on how arguments are created you go into helpers and look single function handling single responsibility.

Write-Hosts announcing parameters is a good thing but I think its better to do that in core extension with something like `Get-PackageParameters -Announce`. Its not the same as you can provide more descriptive information  but in most cases paramters are self describing (do I really need to write _'you want git on path'_ if parameter is `GitOnPath` and is shown during installation in a manner like _User provided parameters: GitOnPath, NoQuickLaunch_.)